nannou/src/draw/renderer/mod.rs

```diff
@@ -974,6 +974,18 @@ impl Renderer {
         let attachment = frame.texture_view();
         let resolve_target = None;
         let mut command_encoder = frame.command_encoder();
+
+        // Ensure background is cleared on first frame if no clear specified.
+        // A previously cleared frame is required for wgpu::LoadOp::Load.
+        if frame.nth() == 0 {
+            let bg_color = draw.state.borrow().background_color;
+            if bg_color.is_none() {
+                if let Ok(mut state) = draw.state.try_borrow_mut() {
+                    state.background_color = Some(Default::default());
+                }
+            }
+        }
```

Comment:
This might sound a little crazy, but I think we should also consider the case where a user might have a transparent window - is it still possible with a setup like this? I guess it might work still as long as `background_color` is set to a zero-alpha value?

Reply:
I verified that if the window is transparent, clearing it with alpha=1. results in a colored background, while alpha=0. results in transparent. So I'll move this code into `to_frame()` and set the alpha based on whether the window is transparent.

Reply:
If you resize the window or restore it later, I'm unsure whether the frame needs to be re-cleared. I noticed the frame &  window counters just continually increase no matter what, so they wouldn't be useful in that situation.

Reply:
> If you resize the window or restore it later, I'm unsure whether the frame needs to be re-cleared

Yeah I think clearing on the first frame is probably enough, it's not super clear that there's a single correct behaviour for most users when it comes to resizing (e.g. we might clear over a long-running additive sketch), so it might be best to leave that case up to the user.

Reply:
I noticed it always loses the old frame data upon resize, because it always allocates a new frame buffer. So I concluded it does need clearing on the first frame of each new framebuffer.

This seems consistent with the recent user-reported issue on Matrix, who said the fullscreen transition happened on about the tenth frame. It only showed uninitialized contents (weird color tint) when it moved to fullscreen.